src/App.js

```diff
@@ -23,15 +23,19 @@ class App extends Component {
     sideBarActiveLink: "",
   };
 
+  components = {
+    // key name: component, route, acceptable queries
+    BugList: [BugList, "bug-list", ""],
+    NewBug: [NewBug, "new-bug", ""],
+    EditBug: [EditBug, "bug-list", "/:bug_id"],
+    AddUser: [AddUser, "add-user", ""],
+    EditProject: [EditProject, "edit-project", ""],
+  };
+
   serverRootUrl() {
     return "http://localhost:4000";
```

Comment:
```suggestion
    return {process.env.ROOT_URL};
```

Reply:
How do you change process.env.ROOT_URL?

I googled for ways to achieve what I've done here and there's multiple ways, the one you suggested i seen people cautioning against for some reason. 

The way i implemented it is one suggestion i seen.

Reply:
i would prefer to go the process.env.ROOT_URL way as then i don't need to pass the root as a prop to each component